front/app/utils/dateUtils.ts

```diff
@@ -249,3 +249,59 @@ export function calculateRoundedEndDate(
   endDate.setMinutes(startDate.getMinutes() + durationInMinutes);
   return endDate;
 }
+
+export const userTimezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+
+// Why do we need this function?
+// The backend sends dates in the format "YYYY-MM-DD" without a time component.
+// When we parse this date in the frontend, it is interpreted as
+// midnight in UTC.
+// This means that if we are west of UTC, e.g. in Brazil,
+// The date will be interpreted as 21:00 the previous day.
+// This function makes sure that the date is always interpreted as midnight in the user's timezone.
+const backendDatestringRegex = /^\d{4}-\d{2}-\d{2}$/;
```

Comment:
We could make this stricter to allow correct numbers for the year, month and day
```suggestion
const backendDatestringRegex = /^\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01])$/;
```